Python/jit.c

```diff
@@ -416,7 +416,13 @@ _PyJIT_Compile(_PyExecutorObject *executor, const _PyUOpInstruction *trace, size
         patches[HoleValue_DATA] = (uint64_t)data;
         patches[HoleValue_EXECUTOR] = (uint64_t)executor;
         patches[HoleValue_OPARG] = instruction->oparg;
+    #if SIZEOF_VOID_P == 8
         patches[HoleValue_OPERAND] = instruction->operand;
+    #else
+        assert(SIZEOF_VOID_P == 4);
+        patches[HoleValue_OPERAND_HI] = instruction->operand >> 32;
```

Comment:
Should `patches` become `uintptr_t` instead of `uint64_t`? (on all platforms).
Given many of the casts are of the form `(uint64_t)some_pointer`, changing them to `(uintptr_t)some_pointer` avoids unnecessary widening on 32 bit platforms.